lib/agentx.c

```diff
@@ -24,6 +24,7 @@
 XREF_SETUP();
 
 DEFINE_HOOK(agentx_enabled, (), ());
+DEFINE_HOOK(bgp_snmp_traps_config_write, (struct vty *vty), (vty));
```

Comment:
This is wrong to define a BGP-related hook in the common library code. Why does BGP trap config depends on `agentx` being enabled anyway? You can use `bgp snmp traps bgp4-mibv2` and it won't be in the running config until you enable `agentx`. It shouldn't work like that.

Reply:
Good catch @idryzhov :+1:

Reply:
taken into account